### PR TITLE
Wegld coverage

### PR DESCRIPTION
**Pull request by andreiblt1304**
Tests fixes after wegld fee logic modifications + more tests

### CODE REVIEW

**Review by alyn509 — APPROVED** [review]

File: enshrine-esdt-safe/tests/enshrine_esdt_safe_blackbox_test.rs [hunk 1]
```diff
@@ -101,34 +150,43 @@ impl EnshrineTestState {
...
-    fn propose_execute_operation(&mut self, has_prefix: bool, error_msg: Option<&str>) {
+    fn propose_execute_operation(
+        &mut self,
+        has_prefix: bool,
+        error_status: Option<ErrorStatus>,
+        tokens: &Vec<TestTokenIdentifier>,
+    ) {
         let (tokens, data) = if has_prefix {
-            self.setup_payments(vec![PREFIX_NFT_TOKEN_ID, FUNGIBLE_TOKEN_ID])
+            self.setup_payments(tokens)
```
Comment (axenteoctavian) [inline enshrine-esdt-safe/tests/enshrine_esdt_safe_blackbox_test.rs:171]: same `self.setup_payments(tokens)` is both on if and else

File: enshrine-esdt-safe/tests/enshrine_esdt_safe_blackbox_test.rs [hunk 2]
```diff
@@ -144,31 +202,11 @@ impl EnshrineTestState {
...
-            .typed(enshrine_esdt_safe_proxy::EnshrineEsdtSafeProxy)
-            .set_header_verifier_address(HEADER_VERIFIER_ADDRESS)
-            .run();
-    }
-
-    fn propose_register_operation(&mut self, has_prefix: bool) {
+    fn propose_register_operation(&mut self, has_prefix: bool, tokens: &Vec<TestTokenIdentifier>) {
         let (tokens, data) = if has_prefix {
-            self.setup_payments(vec![PREFIX_NFT_TOKEN_ID, FUNGIBLE_TOKEN_ID])
+            self.setup_payments(tokens)
```
Comment (axenteoctavian) [inline enshrine-esdt-safe/tests/enshrine_esdt_safe_blackbox_test.rs:207]: same `self.setup_payments(tokens)` is both on if and else

Comment (andreiblt1304) [reply enshrine-esdt-safe/tests/enshrine_esdt_safe_blackbox_test.rs]: fixed

File: enshrine-esdt-safe/tests/enshrine_esdt_safe_blackbox_test.rs (hunk 1 above)
Comment (andreiblt1304) [reply enshrine-esdt-safe/tests/enshrine_esdt_safe_blackbox_test.rs]: fixed

**Review by axenteoctavian — APPROVED** [review]
We also need more tests but you can create them in another PR.
- No test for execute_operation with error
- No test for operation with `opt_transfer_data`
- after propose_execute_operation we need to check tokens are in the receiver wallet
- tests for `deposit` endpoint

**Pull request merged**